plugins/ltac/g_rewrite.mlg

```diff
@@ -99,30 +99,39 @@ ARGUMENT EXTEND rewstrategy
 
     RAW_PRINTED BY { pr_raw_strategy env sigma }
     GLOB_PRINTED BY { pr_glob_strategy env sigma }
+END
 
-  | [ glob(c) ] -> { StratConstr (c, true) }
-  | [ "<-" constr(c) ] -> { StratConstr (c, false) }
-  | [ "subterms" rewstrategy(h) ] -> { StratUnary (Subterms, h) }
-  | [ "subterm" rewstrategy(h) ] -> { StratUnary (Subterm, h) }
-  | [ "innermost" rewstrategy(h) ] -> { StratUnary(Innermost, h) }
-  | [ "outermost" rewstrategy(h) ] -> { StratUnary(Outermost, h) }
-  | [ "bottomup" rewstrategy(h) ] -> { StratUnary(Bottomup, h) }
-  | [ "topdown" rewstrategy(h) ] -> { StratUnary(Topdown, h) }
-  | [ "id" ] -> { StratId }
-  | [ "fail" ] -> { StratFail }
-  | [ "refl" ] -> { StratRefl }
-  | [ "progress" rewstrategy(h) ] -> { StratUnary (Progress, h) }
-  | [ "try" rewstrategy(h) ] -> { StratUnary (Try, h) }
-  | [ "any" rewstrategy(h) ] -> { StratUnary (Any, h) }
-  | [ "repeat" rewstrategy(h) ] -> { StratUnary (Repeat, h) }
-  | [ rewstrategy(h) ";" rewstrategy(h') ] -> { StratBinary (Compose, h, h') }
-  | [ "(" rewstrategy(h) ")" ] -> { h }
-  | [ "choice" ne_rewstrategy_list(h) ] -> { StratNAry (Choice, h) }
-  | [ "old_hints" preident(h) ] -> { StratHints (true, h) }
-  | [ "hints" preident(h) ] -> { StratHints (false, h) }
-  | [ "terms" constr_list(h) ] -> { StratTerms h }
-  | [ "eval" red_expr(r) ] -> { StratEval r }
-  | [ "fold" constr(c) ] -> { StratFold c }
+GRAMMAR EXTEND Gram
+  GLOBAL: rewstrategy;
+  rewstrategy:
+    [ LEFTA
+      [ h = SELF; ";"; h' = rewstrategy_atom -> { StratBinary (Compose, h, h') } ] ]
+  ;
```

Comment:
This seems incorrect, probably should be 2 rules under rewstrategy.  IIUC no way to resolve the SELF.  Probably breaks building the manual and when I try rewrite_strat in CoqIDE it doesn't execute it (no error, doesn't turn green, either).

Reply:
Uh, right. I forgot the `h = rewstrategy_atom` clause in the split. Sorry.